src/io/g15.rs

```diff
@@ -0,0 +1,202 @@
+#![allow(non_upper_case_globals)]
+#![allow(unused_variables)]
+
+use regex::{Regex, Captures};
+use std::{fs};
+
+// A RegMatch struct contains a Regex and the corresponding function to add the result to the G15Player vec
+// if the regex results in a match.
+pub struct RegMatch(Regex, fn(caps: Captures, players: &mut Vec<G15Player>));
+impl RegMatch {
+    pub fn new(reg_str: &str, func: fn(caps: Captures, players: &mut Vec<G15Player>)) -> RegMatch {
+        let /*p*/regers = Regex::new(reg_str).unwrap();
+        RegMatch (regers, func)
+    }
+}
+
+// m_iAmmo[1] integer (32)      --match--> (player idx) (ammo count)
+pub const REGEX_I_AMMO: &str = r#"^m_iAmmo\[(\d+)\]\s+integer\s+\((\d+)\)$"#;
+pub fn parse_ammo(caps: Captures, players: &mut Vec<G15Player>) {
+    let idx = caps[1].parse::<usize>().unwrap();
+    let ammo = caps[2].parse::<u32>().unwrap();
+    players[idx].ammo = ammo;
+}

```

Comment:
A trick with Rust error handling is that you can propagate errors using `?`, so if the function returns an `Option` or `Result` type you can replace `unwrap()` with `?` and it will just propagate the error instead of crashing. The best thing to do would probably be to define your own Error type in this module that implements `From` for each of the error types that could be returned (`ParseIntError`, and another variant for when the index is not valid) and then you can make the functions return a `Result` of that type. Then you can just `?` each time you need to unwrap and we have a way to see later if there were any issues when parsing. If you want to try that and want a hand I'm happy to help, otherwise I can add a commit to do this myself.

Reply:
Sorry for the confusion.

Reply:
Ok turbofish is a really fucking funny name. Anyway - Looking into this more today. Will deliver something more robust and abstractable. Would it be worthwhile at all writing a way to transcribe `G15Player` into some other player object? I.e. a method that accepts one of each and writes the values in `G15Player` into the other object? If so, can you point me to the definition of that other object?

Reply:
No I think just returning the `Vec<G15Player>` should be fine.